container/container.go

```diff
@@ -1,201 +1,523 @@
 package container
 
 import (
+	"crypto/ecdsa"
 	"crypto/sha256"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
 
 	"github.com/google/uuid"
 	"github.com/nspcc-dev/neofs-api-go/v2/container"
 	v2netmap "github.com/nspcc-dev/neofs-api-go/v2/netmap"
 	"github.com/nspcc-dev/neofs-api-go/v2/refs"
 	"github.com/nspcc-dev/neofs-sdk-go/container/acl"
 	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	neofscrypto "github.com/nspcc-dev/neofs-sdk-go/crypto"
+	neofsecdsa "github.com/nspcc-dev/neofs-sdk-go/crypto/ecdsa"
 	"github.com/nspcc-dev/neofs-sdk-go/netmap"
+	subnetid "github.com/nspcc-dev/neofs-sdk-go/subnet/id"
 	"github.com/nspcc-dev/neofs-sdk-go/user"
 	"github.com/nspcc-dev/neofs-sdk-go/version"
 )
 
+// Container represents descriptor of the NeoFS container. Container logically
+// stores NeoFS objects. Container is one of the basic and at the same time
+// necessary data storage units in the NeoFS. Container includes data about the
+// owner, rules for placing objects and other information necessary for the
+// system functioning.
+//
+// Container type instances can represent different container states in the
+// system, depending on the context. To create new container in NeoFS zero
+// instance be created, initialized using Init method and filled using
+// dedicated methods. Once container is saved in the NeoFS network, it can't be
+// changed: containers stored in the system are immutable, and NeoFS is a CAS
+// of containers that are identified by a fixed length value (see cid.ID type).
```

Comment:
What is `CAS of containers`?

Reply:
`Content-addressable storage of container`, do you we need avoid abbreviation?

Reply:
IMO it's worthwhile since `AssertID` checks if the container structure matches the identifier. Without this statement there may be a question about the meaning of this action.